node/src/db/db_handlers.rs

```diff
@@ -0,0 +1,1140 @@
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+
+use crate::{
+    bead::Bead,
+    braid::{consensus_functions, Braid},
+    db::{
+        init_db::init_db, AncestorTimestampTuple, BeadTuple, BraidpoolDBTypes, CohortIdTuple,
+        CohortTuple, InsertTupleTypes, ParentTimestampTuple, RelativeTuple, TransactionTuple,
+    },
+    error::DBErrors,
+};
+use futures::lock::Mutex;
+use num::ToPrimitive;
+use sqlx::{Pool, Sqlite};
+use tokio::sync::{
+    mpsc::{Receiver, Sender},
+    RwLock,
+};
+
+#[derive(Debug)]
+pub struct DBHandler {
+    //Query reciever inherit to handler only
+    receiver: Receiver<BraidpoolDBTypes>,
+    //Shared across tasks for accessing DB after contention using `Mutex`
+    pub db_connection: Arc<Mutex<Pool<Sqlite>>>,
+    local_braid_arc: Arc<RwLock<Braid>>,
+}
+// The DATABASE_URL environment variable must be set at build time to a database which it can prepare queries against; the database does not have to contain any data but must be the same kind (MySQL, Postgres, etc.) and have the same schema as the database you will be connecting to at runtime.
+//A transaction starts with a call to Pool::begin or Connection::begin.
+// A transaction should end with a call to commit or rollback. If neither are called before the transaction goes out-of-scope, rollback is called. In other words, rollback is called on drop if the transaction is still in-progress.
+impl DBHandler {
+    pub async fn new(
+        local_braid_arc: Arc<RwLock<Braid>>,
+    ) -> Result<(Self, Sender<BraidpoolDBTypes>), DBErrors> {
+        log::info!("Initializing Schema for persistant DB");
+        let connection = match init_db().await {
+            Ok(conn) => conn,
+            Err(error) => {
+                log::error!("An error occurred while initializing and establishing connection with local DB {:?}.", error);
+                return Err(DBErrors::ConnectionToDBNotEstablished {
+                    error: error.to_string(),
+                });
+            }
+        };
+        let (db_handler_tx, db_handler_rx) = tokio::sync::mpsc::channel(1024);
+        Ok((
+            Self {
+                receiver: db_handler_rx,
+                db_connection: Arc::new(Mutex::new(connection)),
+                local_braid_arc: local_braid_arc,
+            },
+            db_handler_tx,
+        ))
+    }
+    //Insertion handlers private
+    pub async fn insert_sequential_insert_bead(&self, bead: Bead) -> Result<(), DBErrors> {
+        log::info!("Sequential insertion query received from the query handler");
+        let mut braid_parent_set: HashMap<usize, HashSet<usize>> = HashMap::new();
+        let braid_data = self.local_braid_arc.read().await;
+        //Constructing the parent set
+        for bead in braid_data.beads.iter().enumerate() {
```

Comment:
We're passing a Bead in but actually iterating over parents by grabbing the braid.

This implies that the caller screwed up and didn't insert parents first.

Assume the caller will insert parents and instead fail if the parents aren't already in the DB.